src/sysdolphin/baselib/jobj.c

```diff
@@ -23,3 +27,249 @@ void HSD_JObjCheckDepend(HSD_JObj* jobj)
         break;
     }
 }
+
+void JObjResetRST(HSD_JObj* jobj, HSD_Joint* joint)
+{
+    if (jobj == NULL || joint == NULL) {
+        return;
+    }
+    jobj->rotate.x = joint->rotation.x;
+    jobj->rotate.y = joint->rotation.y;
+    jobj->rotate.z = joint->rotation.z;
+    jobj->scale = joint->scale;
+    jobj->translate = joint->position;
+    if (!(jobj->flags & MTX_INDEP_SRT)) {
+        HSD_JObjSetMtxDirty(jobj);
+    }
+}
+
+void HSD_JObjResetRST(HSD_JObj* jobj, HSD_Joint* joint)
+{
+    if (jobj == NULL || joint == NULL) {
+        return;
+    }
+    JObjResetRST(jobj, joint);
+    if (!(jobj->flags & 0x1000)) {
+        HSD_JObj* child_jobj = jobj->child;
+        HSD_Joint* child_joint = joint->child;
+        while (child_jobj != NULL) {
+            HSD_JObjResetRST(child_jobj, child_joint);
+            child_jobj = child_jobj->next;
+            child_joint = child_joint != NULL ? child_joint->next : NULL;
+        }
+    }
+}
+
+extern char lbl_804D5DB0[8];
+void func_8036EFAC(HSD_JObj* jobj, void (*cb)(HSD_JObj*, void*, u32), void* args)
+{
+    u32 type;
+
+    if (jobj == NULL) {
+        return;
+    }
+    if (jobj->parent == NULL) {
+        __assert(lbl_804D5DB0, 0xAE, "jobj->parent");
+    }
+    type = jobj->parent->child == jobj ? 1 : 2;
+    if (cb != NULL) {
+        cb(jobj, args, type);
+    }
+    if (!(jobj->flags & 0x1000)) {
+        HSD_JObj* child = jobj->child;
+        while (child != NULL) {
+            func_8036EFAC(child, cb, args);
+            child = child->next;
+        }
+    }
+}
+
+void func_8036F0F0(HSD_JObj* jobj, void (*cb)(HSD_JObj*, void*, u32), void* args)
```

Comment:
Should this cb be a typedef?

Reply:
Will do so once it is used elsewhere in the future.